core/templates/pages/contributor-dashboard-admin-page/contributor-dashboard-tables/contributor-admin-stats-table.component.ts

```diff
@@ -232,152 +437,85 @@ export class ContributorAdminStatsTable implements OnInit {
-            if (this.dataSource.length === 0) {
-              this.noDataMessage = 'No statistics to display';
-            }
-          });
-    } else if (this.inputs.activeTab === this.TAB_NAME_QUESTION_REVIEWER) {
-      this.columnsToDisplay = [
-        'chevron',
-        'contributorName',
-        'reviewedQuestionsCount',
-        'lastContributedInDays',
-        'role'
-      ];
-      if (this.checkMobileView()) {
-        this.columnsToDisplay = [
-          'contributorName',
-          'reviewedQuestionsCount',
-          'lastContributedInDays',
-          'role',
-          'chevron'
-        ];
-      }
-      this.ContributorDashboardAdminStatsBackendApiService
-        .fetchContributorAdminStats(
-          this.inputs.filter,
-          this.itemsPerPage,
-          this.nextOffset,
-          AppConstants.CONTRIBUTION_STATS_TYPE_QUESTION,
-          AppConstants.CONTRIBUTION_STATS_SUBTYPE_REVIEW).then(
-          (response) => {
-            this.dataSource = response.stats;
-            this.nextOffset = response.nextOffset;
-            this.more = response.more;
-            this.loadingMessage = '';
-            this.noDataMessage = '';
-            if (this.dataSource.length === 0) {
-              this.noDataMessage = 'No statistics to display';
-            }
-          });
+  getContributionType(activeTab: string): string {
+    let contributionType: string = '';
+
+    switch (activeTab) {
+      case this.TAB_NAME_TRANSLATION_SUBMITTER:
+      case this.TAB_NAME_TRANSLATION_REVIEWER:
+        contributionType = AppConstants.CONTRIBUTION_STATS_TYPE_TRANSLATION;
+        break;
+      case this.TAB_NAME_QUESTION_SUBMITTER:
+      case this.TAB_NAME_QUESTION_REVIEWER:
+        contributionType = AppConstants.CONTRIBUTION_STATS_TYPE_QUESTION;
+        break;
+    }
+
+    return contributionType;
+  }
+
+  getContributionSubType(activeTab: string): string {
+    let contributionSubType: string = '';
+
+    switch (activeTab) {
+      case this.TAB_NAME_TRANSLATION_SUBMITTER:
+      case this.TAB_NAME_QUESTION_SUBMITTER:
+        contributionSubType =
+        AppConstants.CONTRIBUTION_STATS_SUBTYPE_SUBMISSION;
+        break;
+      case this.TAB_NAME_TRANSLATION_REVIEWER:
+      case this.TAB_NAME_QUESTION_REVIEWER:
+        contributionSubType = AppConstants.CONTRIBUTION_STATS_SUBTYPE_REVIEW;
+        break;
+    }
+
+    return contributionSubType;
+  }
+
+  updateColumns(contributionSubType: string): void {
+    if (this.checkMobileView()) {
+      this.columnsToDisplay = ['contributorName'];
+    } else {
+      this.columnsToDisplay = ['chevron', 'contributorName'];
     }
+    if (contributionSubType ===
+        AppConstants.CONTRIBUTION_STATS_SUBTYPE_SUBMISSION) {
+      this.columnsToDisplay.push('recentPerformance', 'overallAccuracy');
+    }
+    this.columnsToDisplay.push(
+      'contributionCount', 'lastContributedInDays', 'role');
+
+    if (this.checkMobileView()) {
```

Comment:
This function doesn't sound like it returns a boolean ... perhaps isMobileView()?

Reply:
Done